website/docs/reference/widgets/iframe.md

```diff
@@ -1,100 +1,205 @@
 
-1. On a blank canvas, drop a new iframe widget.
+```html
+ <script>
+        function sendSelectedTime() {
+            var selectedHour = document.getElementById("hourSelect").value;
+            var selectedMin = document.getElementById("minSelect").value;
+            var selectedAMPM = document.getElementById("ampmSelect").value;
+            var selectedTime = selectedHour + ":" + selectedMin + " " + selectedAMPM;
+            window.parent.postMessage(selectedTime, "*");
+        }
+    </script>
+```
+The JavaScript function, `sendSelectedTime()`, retrieves the selected values from the select elements, concatenates them into a string in the format `"hh:mm AM/PM"`, and then sends that string to the parent window using the `postMessage()` method. This allows the selected time to be passed for further use.
 
-2. Embed a page that's able to send a message with `postMessage()`. In the iframe widget’s settings, copy and paste the following snippet into its **srcDoc** property:
 
-   ```html
-   <input id="messageinput" type="text"></input>
-   <input type="button" onclick="sendMyMessage()" value="SEND" />
-   <script>
-       function sendMyMessage() {
-           const msgText = document.getElementById("messageinput").value;
-           window.parent.postMessage(msgText, "*");
-       }
-   </script>
-   ```
+<figure>
+  <img src="/img/custom-widget-time-picker.gif" style= {{width:"700px", height:"auto"}} alt="Display external website"/>
+  <figcaption align = "center"><i></i></figcaption>
+</figure>
 
-   You’ve created a simple HTML document in the iframe containing a text input, a button, and a script to handle sending the message.
 
-3. Drag and drop a new [Text](text.md) widget onto the canvas, and set its Text property to `{{Iframe1.message}}`.
+## Communication with Iframe
 
-4. Type a string into the iframe’s input box and click the "Send” button. You should see your [Text widget](text.md) update to contain the string that you sent from the iframe.
+Appsmith allows secure communication between different Window objects, including iframes. There are a number of directions in which your apps may need to communicate. Depending on their parent/embed relationship and the message's target recipient, the configuration is a bit different. 
+Refer to this [document](/reference/appsmith-framework/widget-actions/post-message) for information on embedding your applications externally.
 
-:::info
-When a message is received, Appsmith also executes any code or actions set within the iframe’s [`onMessageReceived` ](iframe.md#events)event. Try it out – set an action within the `onMessageReceived` event, and send another message to watch the results.
-:::
 
-### Events
+### From Appsmith to embedded page
 
```

Comment:
PLease add the image from here - https://appsmith-docs-git-iframe-rehaul-get-appsmith.vercel.app/reference/appsmith-framework/widget-actions/post-message#from-appsmith-to-embedded-page